src/serialization/ObjectData.ts

```diff
@@ -338,8 +342,22 @@ export class ObjectDataInput implements DataInput {
         return this.read(pos);
     }
 
-    readByteArray(pos?: number): number[] {
-        return this.readArray<number>(this.readByte, pos);
+    readByteArray(pos?: number): Buffer {
+        const backupPos = this.pos;
+        if (pos !== undefined) {
+            this.pos = pos;
+        }
+        const len = this.readInt();
+        if (len === BitsUtil.NULL_ARRAY_LENGTH) {
+            return null;
```

Comment:
If the `pos` (the argument one) is not `undefined` shouldn't we set back `this.pos` to `backupPos` before returning?. I checked the `readArray` method and it seems it is not doing it also.

By the way, Java side does not expect `pos` argument for `readXArray` methods. I cannot recall whether or not this argument is needed. Maybe we could also check that.

Reply:
> If the `pos` (the argument one) is not `undefined` shouldn't we set back `this.pos` to `backupPos` before returning?. I checked the `readArray` method and it seems it is not doing it also.

We restore `this.pos`, just like it's done in `readArray`. See L356.

Reply:
I meant the early return

```js
if (len === BitsUtil.NULL_ARRAY_LENGTH) {
    // this.pos is corrupt now if the pos is undefined
    return null;
}
```

Reply:
Oups, missed that point. Yes, it should be restored. Let me fix that.

Reply:
Fixed in 71d242795651a102cea7dc54ae94d3d6a806003b